app/src/main/java/com/github/se/cyrcle/ui/report/ReviewReportScreen.kt

```diff
@@ -0,0 +1,184 @@
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.github.se.cyrcle.R
+import com.github.se.cyrcle.model.review.ReviewReport
+import com.github.se.cyrcle.model.review.ReviewReportReason
+import com.github.se.cyrcle.model.review.ReviewViewModel
+import com.github.se.cyrcle.model.user.UserViewModel
+import com.github.se.cyrcle.ui.navigation.NavigationActions
+import com.github.se.cyrcle.ui.theme.atoms.BulletPoint
+import com.github.se.cyrcle.ui.theme.atoms.ConditionCheckingInputText
+import com.github.se.cyrcle.ui.theme.disabledColor
+import com.github.se.cyrcle.ui.theme.molecules.EnumDropDown
+import com.github.se.cyrcle.ui.theme.molecules.TopAppBar
+
+@SuppressLint("StateFlowValueCalledInComposition")
+@Composable
+fun ReviewReportScreen(
+    navigationActions: NavigationActions,
+    userViewModel: UserViewModel,
+    reviewViewModel: ReviewViewModel,
+) {
+  val configuration = LocalConfiguration.current
+  val screenWidth = configuration.screenWidthDp.dp
+  val screenHeight = configuration.screenHeightDp.dp
+  val showDialog = remember { mutableStateOf(false) }
+
+  // Define padding as a percentage of screen dimensions
+  val horizontalPaddingScaleFactor = screenWidth * 0.03f
+  val topBoxHeight = screenHeight * 0.10f // 10% of screen height for top box
+  val verticalPaddingScaleFactor = screenHeight * 0.02f
+
+  // State for report inputs
+  val selectedReason = rememberSaveable { mutableStateOf(ReviewReportReason.IRRELEVANT) }
+  val reviewId = reviewViewModel.selectedReview.value?.uid
+  val reportDescription = rememberSaveable { mutableStateOf("") }
+  val userId = userViewModel.currentUser.value?.public?.userId
+  val context = LocalContext.current
+
+  fun onSubmit() {
+    if (reviewId != null) {
+      val report =
+          ReviewReport(
+              uid = reviewViewModel.getNewUid(),
+              reason = selectedReason.value,
+              userId = userId ?: "",
+              review = reviewId,
+              description = reportDescription.value)
+      reviewViewModel.addReport(report, userViewModel.currentUser.value!!)
+      Toast.makeText(context, "Report added", Toast.LENGTH_SHORT).show()
```

Comment:
Could you add the string to string.xml since it is displayed?

Reply:
Done!